include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,218 @@
+#include "dlaf/factorization/qr.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix/copy.h"
+#include "dlaf/matrix/copy_tile.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/matrix/layout_info.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+namespace eigensolver {
+namespace internal {
+
+template <class T>
+void set_zero(Matrix<T, Device::CPU>& mat) {
+  dlaf::matrix::util::set(mat, [](auto&&) { return static_cast<T>(0.0); });
+}
+
+// Implementation based on:
+// 1. Part of algorithm 6 "LAPACK Algorithm for the eigenvector back-transformation", page 15, PhD thesis
+// "GPU Accelerated Implementations of a Generalized Eigenvalue Solver for Hermitian Matrices with
+// Systematic Energy and Time to Solution Analysis" presented by Raffaele Solcà (2016)
+// 2. Report "Gep + back-transformation", Alberto Invernizzi (2020)
+// 3. Report "Reduction to band + back-transformation", Raffaele Solcà (2020)
+// 4. G. H. Golub and C. F. Van Loan, Matrix Computations, chapter 5, The Johns Hopkins University Press
+template <class T>
+struct BackTransformation<Backend::MC, Device::CPU, T> {
+  static void call_FC(Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+  static void call_FC(comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_c,
+                      Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+};
+
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  constexpr auto Left = blas::Side::Left;
+  constexpr auto Right = blas::Side::Right;
+  constexpr auto Upper = blas::Uplo::Upper;
+  constexpr auto Lower = blas::Uplo::Lower;
+  constexpr auto NoTrans = blas::Op::NoTrans;
+  constexpr auto ConjTrans = blas::Op::ConjTrans;
+  constexpr auto NonUnit = blas::Diag::NonUnit;
+
+  using hpx::util::unwrapping;
+
+  using hpx::execution::parallel_executor;
+  using hpx::resource::get_thread_pool;
+  using hpx::threads::thread_priority;
+
+  auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mv = mat_v.nrTiles().rows();
+  const SizeType nv = mat_v.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+  const SizeType nb = mat_v.blockSize().cols();
+  const SizeType ms = mat_v.size().rows();
+  const SizeType ns = mat_v.size().cols();
+
+  // Matrix T
+  comm::CommunicatorGrid comm_grid(MPI_COMM_WORLD, 1, 1, common::Ordering::ColumnMajor);
+  common::Pipeline<comm::CommunicatorGrid> serial_comm(comm_grid);

```

Comment:
As far as I can tell this is only here because `computeTFactor` expects it. However, no communication will be required, right (since this is the single-node overload)? Would it be worth adding an overload to `computeTFactor` that doesn't require this?

Reply:
@rasolca and I agree with you that it is worth adding that overload! I will do that in a separate PR.